### PR TITLE
Add Azure VM Scale Set

### PR DESCRIPTION
**Pull request by anatheka**
This PR adds
- Azure VM Scale Set discovery with VMs

### CODE REVIEW

File: service/discovery/azure/compute.go [hunk 1]
```diff
@@ -399,6 +479,7 @@ func runtimeInfo(runtime string) (runtimeLanguage string, runtimeVersion string)
 }
 
 // Discover virtual machines
+// TODO(all): Do we want to discover the VMs of a scale set, too?
```
Comment (anatheka) [inline service/discovery/azure/compute.go:482]:
```suggestion
```

File: service/discovery/azure/azure.go [hunk 2]
```diff
@@ -165,9 +165,11 @@ type clients struct {
 	sitesClient *armappservice.WebAppsClient
 
 	// Compute
-	virtualMachinesClient *armcompute.VirtualMachinesClient
-	blockStorageClient    *armcompute.DisksClient
-	diskEncSetClient      *armcompute.DiskEncryptionSetsClient
+	virtualMachinesClient           *armcompute.VirtualMachinesClient
+	virtualMachineScaleSetClient    *armcompute.VirtualMachineScaleSetsClient
```
Comment (lebogg) [inline service/discovery/azure/azure.go:169]: just minor thing to stay consistent: Use plural in the last word before Client

File: service/discovery/azure/compute.go [hunk 3]
```diff
@@ -136,6 +144,78 @@ func (d *azureComputeDiscovery) List() (list []voc.IsCloudResource, err error) {
...
+		d.clients.virtualMachineScaleSetClient.NewListPager,
+		func(res armcompute.VirtualMachineScaleSetsClientListAllResponse) []*armcompute.VirtualMachineScaleSet {
+			return res.Value
+		},
+		func(res armcompute.VirtualMachineScaleSetsClientListResponse) []*armcompute.VirtualMachineScaleSet {
+			return res.Value
+		},
+
+		func(scaleSet *armcompute.VirtualMachineScaleSet) error {
+			r, err := d.discoverVMScaleSetVMs(scaleSet)
```
Comment (lebogg) [inline service/discovery/azure/compute.go:168]: Shouldn't that be `handleXXX` to be consistent?

Comment (lebogg) [reply service/discovery/azure/compute.go:168]: Ok, it seems to be a nested discover-then-handle-function

File: service/discovery/azure/compute.go [hunk 4]
```diff
@@ -436,6 +516,65 @@ func (d *azureComputeDiscovery) discoverVirtualMachines() ([]voc.IsCloudResource
...
+					Region: *vm.Location,
+				},
+				labels(vm.Tags),
+				voc.ResourceID(parent),
+				voc.VirtualMachineType,
+				vm,
+			),
+		},
+
+		AutomaticUpdates: automaticUpdatesScaleSet(vm.Properties.OSProfile),
```
Comment (lebogg) [inline service/discovery/azure/compute.go:536]: Testing with the testbed on Azure, I get a nil pointer dereference here because `vm.Properties` is nil

**Review by lebogg — CHANGES_REQUESTED** [review]
One nil pointer deference occured

File: service/discovery/azure/compute.go (hunk 3 above)
Comment (anatheka) [reply service/discovery/azure/compute.go:168]: No, because first the VMs in the scale set must be discovered and then the handleX method is called.

**Pull request merged**